src/__tests__/__snapshots__/index.test.js.snap

```diff
@@ -33,6 +33,21 @@ Object {
 }
 `;
 
+exports[`denormalize denormalizes recursive dependencies 1`] = `
+Object {
+  "title": "Weekly report",
+  "user": Object {
+    "reports": Array [
+      Object {
+        "title": "Weekly report",
+        "user": Object {},
```

Comment:
I would expect this to be the user's ID, not an empty object. I don't think this is actually working the way you want.